src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -154,24 +127,60 @@ private void updateSuccessMessage(StringBuilder successMessage, Person updatedPe
         successMessage.append(Messages.format(updatedPerson));
     }
 
-    private void updateFinalMessage(StringBuilder finalMessage, Set<Tag> duplicateTags, Set<Tag> missingTags) {
+    private void updateFinalMessage() {
         if (!finalMessage.isEmpty()) {
             finalMessage.append("\n");
         }
         if (!missingTags.isEmpty()) {
             finalMessage.append(MESSAGE_TAG_NOT_CREATED)
-                    .append(missingTags.stream()
-                            .map(Tag::toString)
-                            .collect(Collectors.joining(", ")));
+                    .append(missingTags.stream().map(Tag::toString).collect(Collectors.joining(", ")));
         }
         if (!duplicateTags.isEmpty()) {
             if (!missingTags.isEmpty()) {
                 finalMessage.append("\n");
             }
             finalMessage.append(MESSAGE_DUPLICATE_TAG)
-                    .append(duplicateTags.stream()
-                            .map(Tag::toString)
-                            .collect(Collectors.joining(", ")));
+                    .append(duplicateTags.stream().map(Tag::toString).collect(Collectors.joining(", ")));
+        }
+    }
```

Comment:
Should the happy paths be nested? Maybe it would be better to have the function simply return if duplicateTags.isEmpty() etc

Reply:
Good catch!